src/DependencyInjection/MongoDbBundleExtension.php

```diff
@@ -76,7 +79,18 @@ private function defineConnectionFactory(): void
         $this->containerBuilder->setDefinition('mongo.connection_factory', $factoryDefinition);
     }
 
-    private function defineConnections(array $connections): void
+    private function defineDriverOptionsFactory()
+    {
+        $factoryDefinition = new Definition(DriverOptionsFactory::class, []);
+        $factoryDefinition->setPublic(false);
+
+        $this->containerBuilder->setDefinition('mongo.driver_options_factory', $factoryDefinition);
```

Comment:
Why this instead of `services.xml`?

Reply:
I think I get your idea now how to implement driver options. I would do changes iun my code as soon as posible and send a new PR. Thanks for your review and comments!!!!!

Reply:
new changes in the PR @ilario-pierbattista , check it when you can.